assets/home/sections/quick-links.js

```diff
@@ -2,21 +2,160 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
+import { useState, useRef, useEffect } from '@wordpress/element';
+import { decodeEntities } from '@wordpress/html-entities';
 
 /**
  * Internal dependencies
  */
 import Section from '../section';
+import { Grid, Col } from '../grid';
+import Link from '../link';
 
 /**
- * Quick Links section component.
+ * Component to Install Demo Course. Invoked when clicking on a link pointing to "sensei://install-demo-course".
+ *
+ * @param {Object}   props        Component props.
+ * @param {Function} props.remove Function to call to remove the item from the Quick Links Column.
+ * @return {string} A message to the user with the status of the installation.
+ */
+function InstallDemoCourse( { remove } ) {
+	const [ title, setTitle ] = useState( 'Installing' );
+	const timer = useRef( 0 );
+	const installedMessage = 'Installed!';
+	const clear = () => {
+		if ( timer.current ) {
+			clearTimeout( timer.current );
+		}
+		timer.current = 0;
+	};
+	useEffect( () => {
+		if ( null === title ) {
+			remove();
+		}
+	}, [ title, remove ] );
+	useEffect( () => {
+		const run = () => {
+			setTitle( ( oldTitle ) => {
+				if ( installedMessage === oldTitle ) {
+					clear();
+					return null;
+				}
+				if ( ! oldTitle.includes( '...' ) ) {
+					timer.current = setTimeout( run, 500 );
+					return oldTitle + '.';
+				}
+				timer.current = setTimeout( run, 2000 );
+				return installedMessage;
+			} );
+		};
+		clear();
+		timer.current = setTimeout( run, 500 );
+	}, [] );
+	return title;
+}
+
+const quickLinksSpecialMapping = {
+	'sensei://install-demo-course': InstallDemoCourse,
+};
+
+/**
+ * Component that shows a Quick Link.
+ *
+ * @param {Object} props      Component props.
+ * @param {Object} props.item The item to show.
  */
-const QuickLinks = () => (
-	<Section title={ __( 'Quick Links', 'sensei-lms' ) }>
+const QuickLink = ( { item } ) => {
+	let { url, title } = item;
+
+	let onClick = null;
+	const [ showComponent, setShowComponent ] = useState( false );
+	const [ remove, setRemove ] = useState( false );
+
+	const Component = quickLinksSpecialMapping[ url ];
+	if ( !! Component ) {
+		url = '#';
+		onClick = ( e ) => {
+			e.preventDefault();
+			setShowComponent( true );
+		};
+	}
+	if ( remove ) {
+		return null;
+	}
+	return (
+		<li>
+			{ showComponent ? (
+				<Component
+					restoreLink={ () => setShowComponent( false ) }
+					remove={ () => setRemove( true ) }
+				/>
+			) : (
+				<Link
+					url={ url }
```

Comment:
@aaronfc was checking if `decodeEntities` would also be needed for the `url`. I'm just mentioning him here to check his conclusion and confirm if we also need to do it here. ;)

Reply:
From my tests, URLs MIGHT need to be encoded using entities if they're generated by WordPress in `display` mode - which is the default, unfortunately. A great example of this is the `get_edit_post_link` function: https://developer.wordpress.org/reference/functions/get_edit_post_link/

However, if you inform any other value for the mode, it encodes properly as a URL. So I don't think we need to have `decodeEntities` called here.